swarm/src/connection.rs

```diff
@@ -750,6 +765,27 @@ mod tests {
             StreamUpgradeError::Timeout
         ))
     }
+    
+    #[test]
+    fn test_idle_timeout() {
+        // Create a custom idle timeout
+        let idle_timeout = Duration::from_secs(5);
+
+        let mut connection = Connection::new(
+            StreamMuxerBox::new(PendingStreamMuxer),
+            ConfigurableProtocolConnectionHandler::default(),
+            None,
+            0,
+            Some(idle_timeout),
+        );
+
+        // Create a mock context and pin the connection
+        let mut cx = Context::from_waker(futures::task::noop_waker_ref());
+        let connection = Pin::new(&mut connection);
+
+        let poll_result = connection.poll(&mut cx);
+        assert!(poll_result.is_pending());
+    }
```

Comment:
This test should cover that after 5 seconds, we shut the connection down.

You can use `dummy::ConnectionHandler` which never does anything.